> It breaks webkitpy tests :(

```
Traceback (most recent call last):
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/webkitpy/test/main.py", line 45, in <module>
    from webkitpy.results.upload import Upload
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/webkitpy/results/upload.py", line 28, in <module>
    import requests
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 660, in find_module
    cls.install(name)
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 635, in install
    return all([to_install.install() for to_install in packages])
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 305, in install
    raise ValueError('No archives for {}-{} found'.format(self.pypi_name, self.version))
ValueError: No archives for requests-2.31 found
```

And:

```
Traceback (most recent call last):
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/webkitpy/test/main.py", line 45, in <module>
    from webkitpy.results.upload import Upload
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/webkitpy/results/upload.py", line 28, in <module>
    import requests
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 963, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 906, in _find_spec
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 646, in find_spec
    loader = cls.find_module(fullname, path=path)
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 660, in find_module
    cls.install(name)
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 635, in install
    return all([to_install.install() for to_install in packages])
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 635, in <listcomp>
    return all([to_install.install() for to_install in packages])
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 281, in install
    AutoInstall.install('setuptools')
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 635, in install
    return all([to_install.install() for to_install in packages])
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 635, in <listcomp>
    return all([to_install.install() for to_install in packages])
  File "/Volumes/Data/worker/WebKitPy-Tests-EWS/build/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py", line 305, in install
    raise ValueError('No archives for {}-{} found'.format(self.pypi_name, self.version))
ValueError: No archives for setuptools-68.1.2 found
```

These errors do not happen for me. The EWS seem to be unable to see the latest releases from pypi? I do not know what is wrong.

Unfortunately without these changes `git-webkit` is completely broken. I'll probably need help to finish this.

--- a/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py
+++ b/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py
@@ -53,12 +53,17 @@ version = Version(0, 16, 3)
 from webkitcorepy.autoinstall import Package, AutoInstall
 if sys.version_info > (3, 0):
     AutoInstall.register(Package('mock', Version(4)))
-    AutoInstall.register(Package('setuptools', Version(56, 0, 0)))
 else:
     AutoInstall.register(Package('mock', Version(3, 0, 5)))
-    AutoInstall.register(Package('setuptools', Version(44, 1, 1)))
     if platform.system() == 'Windows':
         AutoInstall.register(Package('win_inet_pton', Version(1, 1, 0), pypi_name='win-inet-pton'))
+
+if sys.version_info >= (3, 8):
+    AutoInstall.register(Package('setuptools', Version(68, 1, 2)))
+elif sys.version_info >= (3, 0):
+    AutoInstall.register(Package('setuptools', Version(56, 0, 0)))
+else:
+    AutoInstall.register(Package('setuptools', Version(44, 1, 1)))
 
 if sys.version_info >= (3, 6):
     AutoInstall.register(Package('certifi', Version(2022, 12, 7)))
@@ -77,12 +82,22 @@ else:
     AutoInstall.register(Package('packaging', Version(20, 4)))
 
 AutoInstall.register(Package('pyparsing', Version(2, 4, 7)))
-AutoInstall.register(Package('requests', Version(2, 24)))
+
+if sys.version_info >= (3, 12):
+    AutoInstall.register(Package('requests', Version(2, 31, 0)))
+else:
+    AutoInstall.register(Package('requests', Version(2, 24)))
+
 AutoInstall.register(Package('setuptools_scm', Version(5, 0, 2), pypi_name='setuptools-scm'))
 AutoInstall.register(Package('socks', Version(1, 7, 1), pypi_name='PySocks'))
 AutoInstall.register(Package('six', Version(1, 15, 0)))
 AutoInstall.register(Package('tblib', Version(1, 7, 0)))
-AutoInstall.register(Package('urllib3', Version(1, 25, 10)))
+
+if sys.version_info >= (3, 12):
+    AutoInstall.register(Package('urllib3', Version(2, 0, 4)))
+else:
+    AutoInstall.register(Package('urllib3', Version(1, 25, 10)))
+
 AutoInstall.register(Package('wheel', Version(0, 35, 1)))
 AutoInstall.register(Package('whichcraft', Version(0, 6, 1)))
 AutoInstall.register(Package('cffi', Version(1, 15, 1)))

